#! /bin/sh /usr/share/dpatch/dpatch-run
## 021_dbd_mysql_autofoo.dpatch by <peter@p12n.org>
##
## DP: Use AC_CHECK_HEADERS rather than AC_CHECK_HEADER, so that
## DP: autoheader is smart enough to notice HAVE_MYSQL_MYSQL_H.
## DP: Also fix (remove) a bogus '-I' that otherwise pollutes
## DP: apr-1-config.

@DPATCH@
Index: build/dbd.m4
--- a/build/dbd.m4
+++ b/build/dbd.m4
@@ -77,37 +77,24 @@
   ], [
     apu_have_mysql=0
     if test "$withval" = "yes"; then
-      AC_CHECK_HEADER(mysql.h, AC_CHECK_LIB(mysqlclient_r, mysql_init, [apu_have_mysql=1]))
-      if test "$apu_have_mysql" == "0"; then
-        AC_CHECK_HEADER(mysql/mysql.h, AC_CHECK_LIB(mysqlclient_r, mysql_init, [apu_have_mysql=1]))
-        if test "$apu_have_mysql" != "0"; then
-          APR_ADDTO(APRUTIL_INCLUDES, [-I$withval/include/myql])
-        fi
-      fi
-    elif test "$withval" = "no"; then
-      apu_have_mysql=0
-    else
+      AC_CHECK_HEADERS([mysql.h mysql/mysql.h],
+                       AC_CHECK_LIB(mysqlclient_r, mysql_init, [apu_have_mysql=1]))
+    elif test "$withval" != "no"; then
       CPPFLAGS="-I$withval/include"
       LIBS="-L$withval/lib "
 
       AC_MSG_NOTICE(checking for mysql in $withval)
-      AC_CHECK_HEADER(mysql.h, AC_CHECK_LIB(mysqlclient_r, mysql_init, [apu_have_mysql=1]))
+      AC_CHECK_HEADERS([mysql.h mysql/mysql.h],
+                       AC_CHECK_LIB(mysqlclient_r, mysql_init, [apu_have_mysql=1]))
       if test "$apu_have_mysql" != "0"; then
         APR_ADDTO(APRUTIL_LDFLAGS, [-L$withval/lib])
         APR_ADDTO(APRUTIL_INCLUDES, [-I$withval/include])
       fi
-
-      if test "$apu_have_mysql" != "1"; then
-        AC_CHECK_HEADER(mysql/mysql.h, AC_CHECK_LIB(mysqlclient_r, mysql_init, [apu_have_mysql=1]))
-        if test "$apu_have_mysql" != "0"; then
-          APR_ADDTO(APRUTIL_INCLUDES, [-I$withval/include/mysql])
-          APR_ADDTO(APRUTIL_LDFLAGS, [-L$withval/lib])
-        fi
-      fi
     fi
   ], [
     apu_have_mysql=0
-    AC_CHECK_HEADER(mysql.h, AC_CHECK_LIB(mysqlclient_r, mysql_init, [apu_have_mysql=1]))
+    AC_CHECK_HEADERS([mysql.h mysql/mysql.h],
+                     AC_CHECK_LIB(mysqlclient_r, mysql_init, [apu_have_mysql=1]))
   ])
 
   AC_SUBST(apu_have_mysql)
